src/lib/components/TreeList.svelte

```diff
@@ -2,10 +2,22 @@
   import { cls } from '$lib/utils/styles';
   import { getComponentTheme } from './theme';
 
+
   type Node = { id: number; name: string; level: number; children: Node[] };
 
+  /**
+   * An array of nodes containing custom metadata and an array of child nodes
+   * @param {Array} nodes - An array of Node Objects each containing a children: Node[] attribute.
```

Comment:
I ment to link to the jsdoc `@type` docs - https://jsdoc.app/tags-type.html

Also, if we needed to do per-property comments, I think we would need to use this syntax...

```js
/**
 * @typedef PropertiesHash
 * @type {object}
 * @property {string} id - an ID.
 * @property {string} name - your name.
 * @property {number} age - your age.
 */

/** @type {PropertiesHash} */
var props;
```

but I'm not sure it's worth it over just...

```js
/** @type {{ id: string; name: string; age: number }} */
```

as I don't think it's handled by Sveld (or at least we have a way to show it in the docs).  It could improve VSCode developer experience though (but haven't tested).

Thanks again for tackling this.